pandas/core/ops/common.py

```diff
@@ -116,10 +117,21 @@ def _maybe_match_name(a, b):
     a_has = hasattr(a, "name")
     b_has = hasattr(b, "name")
     if a_has and b_has:
-        if a.name == b.name:
-            return a.name
-        else:
-            # TODO: what if they both have np.nan for their names?
+        try:
+            if a.name == b.name:
```

Comment:
are we somehow raising on NA comparing to a string?

Reply:
if either is pd.NA then this raises TypeError